### PR TITLE
[Ascend] fuj / optimize bmm and triu

### PR DESCRIPTION
**Pull request by jingguo-st**
## Motivation
optimize impl to avoid redundant dispatch for performance:
1. add bmm schema
2. add tiru_ and triu schema

### CODE REVIEW

File: dipu/scripts/autogen_diopi_wrapper/diopi_functions.yaml [hunk 1]
```diff
@@ -1514,6 +1514,15 @@
 - schema: "cos_(Tensor(a!) self) -> Tensor(a!)"
   interface: diopiCosInp(ctx, self)
 
+- schema: "bmm(Tensor self, Tensor mat2) -> Tensor"
+  custom_fallback: True
+  custom_code_at_the_beginning: |
+    TORCH_CHECK(self.sizes().size() == 3, "Tensor self shoud be 3-D tensor.");
+    TORCH_CHECK(mat2.sizes().size() == 3, "Tensor mat2 shoud be 3-D tensor.");
+    TORCH_CHECK(self.sizes()[0] == mat2.sizes()[0], "Tensor batch size of self and mat2 shoud be equal.");
+    auto out = nodispatch::empty({self.sizes()[0], self.sizes()[1], mat2.sizes()[2]}, self.options());
```
Comment (zhaoguochun1995) [inline dipu/scripts/autogen_diopi_wrapper/diopi_functions.yaml:1523]: self.sizes()[0]-> self.size(0)

File: dipu/torch_dipu/csrc_dipu/aten/ops/CustomFallbackFunctions.hpp [hunk 2]
```diff
@@ -392,6 +392,15 @@ static at::Tensor& custom_fallback_dipu_addmm_out(
   return out;
 }
 
+static at::Tensor custom_fallback_dipu_bmm(const at::Tensor& self,
```
Comment (yewentao256) [inline dipu/torch_dipu/csrc_dipu/aten/ops/CustomFallbackFunctions.hpp:395]: 为什么这里要fallback呢？

Comment (jingguo-st) [reply dipu/torch_dipu/csrc_dipu/aten/ops/CustomFallbackFunctions.hpp:395]: "bmm" not implemented for 'Half'

Comment (jingguo-st) [reply dipu/torch_dipu/csrc_dipu/aten/ops/CustomFallbackFunctions.hpp:395]: - schema: "bmm(Tensor self, Tensor mat2) -> Tensor"
  custom_fallback: True
如果不加这个实现，custom_fallback: True 会编译不过去

**Review by yewentao256 — APPROVED** [review]

**Review by zhaoguochun1995 — APPROVED** [review]

**Review by zhaoguochun1995 — APPROVED** [review]

**Pull request merged**